trees/treeconfig.go

```diff
@@ -0,0 +1,185 @@
+package trees
```

Comment:
We should call it just `tree`. And we should move it to the internal package: `internal/tree`